src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -389,7 +389,7 @@ impl ToInteriorKind for mc::InteriorKind {
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
 pub enum LoanPathElem {
     LpDeref(mc::PointerKind),    // `*LV` in README.md
-    LpInterior(InteriorKind),    // `LV.f` in README.md
+    LpInterior(Option<ast::DefId>, InteriorKind),    // `LV.f` in README.md
```

Comment:
Can you add a comment here to explain what the DefId is for and when you should/should not have one